source/grid/grid_generator.cc

```diff
@@ -8782,11 +8780,28 @@ namespace GridGenerator
       }
     else
       {
-        AssertThrow(colorize == false, ExcNotImplemented());
+        AssertThrow(false, ExcNotImplemented());
       }
 
     // actually create triangulation
     tria.create_triangulation(vertices, cells, SubCellData());
+
+    if (colorize)
+      {
+        // to colorize, run through all faces of all cells and set boundary
+        // indicator to the correct value if it was 0.
+
+        // use a large epsilon to compare numbers to avoid roundoff problems.
+        double epsilon = 10;
+        for (unsigned int i = 0; i < dim; ++i)
+          epsilon = std::min(epsilon,
+                             0.01 * (std::abs(p2[i] - p1[i]) / repetitions[i]));
```

Comment:
This is not the right way to approach this. `p2[i] - p1[i]` is a quantity that depends on the magnitude of the points chosen as inputs, so it has physical units. But `eps=10` does not. You need to start with an eps that has the right order of magnitude. Alternatively, you could start with `eps=infinity`, i.e., `eps=std::numeric_limits<double>::max()`.